### ISSUE TITLE
Inconsistent sub -> method / method -> sub implementation

### ISSUE THREAD

**Issue by zoffixznet** [rakudo/rakudo#2309]
Just writing up docs for some `Cool` methods and noticing inconsistency in how they have to be described.

For example, `Cool.substr` [coercers the invocant](https://github.com/rakudo/rakudo/blob/b22edce40/src/core/Cool.pm6#L154-L155) and then delegates the method call. However, if [you look at `Cool.printf`/`Cool.sprintf`](https://github.com/rakudo/rakudo/blob/b22edce40/src/core/Cool.pm6#L220-L221), then we don't coerce anything at all and instead call the subform.

This sort of differences affect users who are trying to extend or subclass core types and we probably should be more consistent with what we do.

**Comment by JJ** [comment]
This kind of issues should probably go to the problem-solving repository now, right?

**Comment by lizmat** [comment]
Possibly.  As far as I know, the intent has always been that subs refer to the method version.  In fact, a few months ago, I ported a bunch of them in `Str` if I remember correctly.

**Comment by jnthn** [comment]
> As far as I know, the intent has always been that subs refer to the method version.

This is correct. Further, the idea of `Cool` is that it coerces and then calls the method on the coerced result.
